src/geo/projection/globe_util.js

```diff
@@ -499,17 +499,40 @@ export function globeUseCustomAntiAliasing(painter: Painter, context: Context, t
     return transitionT === 0.0 && !useContextAA && !disabled && hasStandardDerivatives;
 }
 
-export function getGridMatrix(id: CanonicalTileID, bounds: LngLatBounds, latitudinalLod: number): Array<number> {
+export function getGridMatrix(id: CanonicalTileID, bounds: LngLatBounds, latitudinalLod: number, worldSize: number): Array<number> {
     const n = bounds.getNorth();
     const s = bounds.getSouth();
     const w = bounds.getWest();
     const e = bounds.getEast();
-    const S = 1.0 / GLOBE_VERTEX_GRID_SIZE;
-    const x = (e - w) * S;
-    const latitudinalSubdivs = GLOBE_LATITUDINAL_GRID_LOD_TABLE[latitudinalLod];
-    const y = (s - n) / latitudinalSubdivs;
-    const tileZoom = 1 << id.z;
-    return [0, x, tileZoom, y, 0, id.y, n, w, S];
+
+    // Construct transformation matrix for converting tile coordinates into LatLngs
+    const tiles = 1 << id.z;
+    const tileWidth = e - w;
+    const tileHeight = n - s;
+    const toUv = 1.0 / GLOBE_VERTEX_GRID_SIZE;
+    const x = tileWidth * toUv;
+    const y = -tileHeight / GLOBE_LATITUDINAL_GRID_LOD_TABLE[latitudinalLod];
```

Comment:
how about `toUvX` and  we could also refactor `1 / GLOBE_LATITUDINAL_GRID_LOD_TABLE(..)` to `toUvY`

Reply:
I think it's unnecessary to have a separate variable `toUv` anymore